src/pokemons/pokemons.controller.ts

```diff
@@ -1,30 +1,60 @@
-import { Controller, Get, Query } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Logger,
+  NotFoundException,
+  Query,
+} from '@nestjs/common';
 import { PokemonsService } from './pokemons.service';
 import { FindPokemonsDto } from './dtos/findPokemonsDto';
-import { FILTER_PROPERTY_NAME, SORT_FIELD_MAPPING, SORT_ORDER_MAPPING } from './constants';
+import {
+  FILTER_PROPERTY_NAME,
+  SORT_FIELD_MAPPING,
+  SORT_ORDER_MAPPING,
+} from './constants';
 
 @Controller('pokemons')
 export class PokemonsController {
   constructor(private readonly pokemonsService: PokemonsService) {}
 
   @Get()
   async findAll(@Query() query: FindPokemonsDto) {
-    const { search, isOwned, sortField, sortOrder, page, itemsPerPage } = query;
+    const { search, isOwned, sortField, sortOrder, startIndex, limit } = query;
+
     const filters: any = {};
-    if (search) filters[FILTER_PROPERTY_NAME] = { $regex: search, $options: 'i' };
-    if (isOwned) filters.isOwned = isOwned === 'true';
+    if (search)
+      filters[FILTER_PROPERTY_NAME] = { $regex: search, $options: 'i' };
+    if (isOwned) filters['isOwned'] = isOwned === 'true';
 
-    const sort = { [SORT_FIELD_MAPPING[sortField] ]: SORT_ORDER_MAPPING[sortOrder] };
+    const sort = {
+      [SORT_FIELD_MAPPING[sortField]]: SORT_ORDER_MAPPING[sortOrder],
+    };
 
-    return this.pokemonsService.getFilteredPokemons(
-      filters,
-      sort,
-      page,
-      itemsPerPage,
-    );
+    Logger.log('Getting filtered Pokemon');
+    try {
+      const results = await this.pokemonsService.getFilteredPokemons(
+        filters,
+        sort,
+        startIndex,
+        limit,
+      );
+      results.data.length === 0
+        ? Logger.log(`No Pokemons were found`)
+        : Logger.log(`Successfully retrieved ${results.total} results`);
+      return results;
+    } catch (error) {
+      Logger.error('Could not retrieve Pokemon', error.stack);
+      throw error;
```

Comment:
```suggestion
      const errorMessage = 'Failed to retrieve pokemons'
      Logger.error(errorMessage, error.stack);
      throw new Error(errorMessage + error);
```

I am actually not sure about the error.stack in terms of security - I'll check it out